src/main/java/bookstore/service/UserServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package bookstore.service;
+
+import bookstore.dto.user.UserRegistrationRequest;
+import bookstore.dto.user.UserResponseDto;
+import bookstore.exception.RegistrationException;
+import bookstore.mapper.UserMapper;
+import bookstore.model.User;
+import bookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequest request) throws RegistrationException {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("Unable to complete registration");
+        }
+        User user = new User();
```

Comment:
Let's create method in UserMapper and use it here 
User user = mapper.toUser(request);